samples/onnx_accelerator_sample1/onnxacceleratorsampleone/with_ggv2/components/aws.samples.windturbine.detector/edge_application.py

```diff
@@ -0,0 +1,189 @@
+    # variable containing the decompressed path to the model
+    sess = None
+    try:
+        sess = ort.InferenceSession(args.model_path)
+    except Exception as e:
+        logging.error(e)
+        exit()
+
+    cloud_connector = turbine.CloudConnector()
+    
+    # Some constants used for data prep + compare the results
+    file_path = os.path.dirname(__file__)
+    thresholds = np.load(os.path.join(file_path, 'statistics/thresholds.npy'))
+    raw_std = np.load(os.path.join(file_path, 'statistics/raw_std.npy'))
+    mean = np.load(os.path.join(file_path, 'statistics/mean.npy'))
+    std = np.load(os.path.join(file_path, 'statistics/std.npy'))
+    
+    try:
+        while True:
+            cloud_connector.publish_logs(tokens_q.get())
+
+            if q.qsize() <= MIN_NUM_SAMPLES:
+                if q.qsize() % 10 == 0:
+                    logging.info('Buffering %d/%d... please wait' % (q.qsize(), MIN_NUM_SAMPLES))
+                    time.sleep(1)
+                # buffering
+                continue
+
+            # prep the data for the model
+            li = list(q.queue)
+            data = np.array(li) # create a copy
+            q.get() # remove the oldest sample            
+            data = np.array([turbine.wavelet_denoise(data[:,i], raw_std[i], 'db6') for i in range(NUM_FEATURES)])
+            data = data.transpose((1,0))
+            data -= mean
+            data /= std
+            data = data[-(TIME_STEPS+STEP):]
+
+            x = turbine.create_dataset(data, TIME_STEPS, STEP)
+            x = np.transpose(x, (0, 2, 1)).reshape(x.shape[0], NUM_FEATURES, 10, 10).astype(np.float32)
+
+            # Now we can run our model using the loaded data
+            # The run command lets you specify which outputs you want to get returned. it only has one output.
+            ptemp = sess.run(None, {"input": x})
+
+            # We are converting the prediction output to a numpy array so that we can convert it into
+            # something human readable
+            p = np.asarray(ptemp[0])
+
+            a = x.reshape(x.shape[0], NUM_FEATURES, 100).transpose((0,2,1))
+            b = p.reshape(p.shape[0], NUM_FEATURES, 100).transpose((0,2,1))
+            
+            # check the anomalies
+            pred_mae_loss = np.mean(np.abs(b - a), axis=1).transpose((1,0))
+            values = np.mean(pred_mae_loss, axis=1)
+            anomalies = (values > thresholds)
+            
+            # publish data to visualize in dashboard
+            ts = "%s+00:00" % datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]
+            cloud_connector.publish_inference(anomalies.astype(np.float32), values.astype(np.float32), model_name, model_version, ts)
+
+            if anomalies.any():
+                logging.info("Anomaly detected: %s" % anomalies)
+            else:
+                logging.info("Ok")
+
+            time.sleep(PREDICTIONS_INTERVAL)
+    except KeyboardInterrupt as e:
```

Comment:
Keyboard Interrupts cannot be thrown in a GG component. It is best practice to create a signal handler on SIGINT | SIGTERM and clean up the resource before shutting down. This will speed up the shutdown of the component